nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -697,6 +697,7 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
                     final RecordSchema recordSchema = currentRecord.getSchema();
                     final Map<String, ColumnDescription> columns = tableSchema.getColumns();
 
+                    int deleteIndex = 0;
```

Comment:
Is there a reason for introducing a special delete index as opposed to basing the parameter index off the current value of `i` in the loop?

Reply:
Yeah that's the crux of the change, if the column is nullable you want to skip setting the second parameter, but using `i` will always advance the index whether the parameter was set or not.

Reply:
Thanks for clarifying, that makes sense.